File: src/main/java/ru/reosfire/WMLib/Yaml/Default/Gui/SwitchConfig.java
package ru.reosfire.WMLib.Yaml.Default.Gui;

import org.bukkit.configuration.ConfigurationSection;
import ru.reosfire.WMLib.Yaml.Default.Wrappers.ItemConfig;
import ru.reosfire.WMLib.Yaml.YamlConfig;

import java.util.List;

public class SwitchConfig extends YamlConfig
{
    public final int Index;
    public final List<ItemConfig> Items;
    public final Integer CoolDown;
    public SwitchConfig(ConfigurationSection configurationSection)
    {
        super(configurationSection);
        Index = getInt("Index");
        Items = getNestedConfigs(ItemConfig::new, "Items");
        CoolDown = getInt("CoolDown", 0);
    }
}

File: src/main/java/ru/reosfire/wmlib/yaml/common/gui/SwitchConfig.kt
package ru.reosfire.wmlib.yaml.common.gui

import org.bukkit.configuration.ConfigurationSection
import ru.reosfire.wmlib.yaml.YamlConfig
import ru.reosfire.wmlib.yaml.common.wrappers.ItemConfig

class SwitchConfig(configurationSection: ConfigurationSection?) : YamlConfig(configurationSection) {
    val index: Int = getInt("Index")
    val items: List<ItemConfig> = getNestedConfigs("Items") { ItemConfig(it) }
    val cooldown: Int = getInt("CoolDown", 0)
}